src/tike/ptycho/ptycho.py

```diff
@@ -309,19 +309,41 @@ def reconstruct(
 def _rescale_obj_probe(operator, comm, data, psi, scan, probe, num_batch):
     """Keep the object amplitude around 1 by scaling probe by a constant."""
 
-    i = batch_indicies(data.shape[-3], num_batch, use_random=True)[0]
+    def _get_rescale(data, psi, scan, probe, num_batch, operator):
+        i = batch_indicies(data.shape[-3], num_batch, use_random=True)[0]
+
+        intensity, _ = operator._compute_intensity(data[..., i, :, :], psi,
+                                                   scan[..., i, :], probe)
+
+        n1 = np.linalg.norm(np.ravel(np.sqrt(data[..., i, :, :])))**2
+        n2 = np.linalg.norm(np.ravel(np.sqrt(intensity)))**2
+
+        return n1, n2
 
-    intensity, _ = operator._compute_intensity(data[..., i, :, :], psi,
-                                               scan[..., i, :], probe)
+    n1, n2 = zip(*comm.pool.map(
+        _get_rescale,
+        data,
+        psi,
+        scan,
+        probe,
+        num_batch = num_batch,
+        operator=operator,
+    ))
+
+    if comm.use_mpi:
+        n1 = np.sqrt(comm.Allreduce_reduce(n1, 'cpu'))
+        n2 = np.sqrt(comm.Allreduce_reduce(n2, 'cpu'))
+    else:
+        n1 = np.sqrt(comm.reduce(n1, 'cpu'))
+        n2 = np.sqrt(comm.reduce(n2, 'cpu'))
 
-    rescale = (np.linalg.norm(np.ravel(np.sqrt(data[..., i, :, :]))) /
-               np.linalg.norm(np.ravel(np.sqrt(intensity))))
+    rescale = n1 / n2
 
     logger.info("object and probe rescaled by %f", rescale)
 
-    probe *= rescale
+    probe[0] *= rescale
```

Comment:
This only rescales the probe for the zeroth GPU? What happens when there are multiple GPUs on each node?

Reply:
In your original implementation, you give probe[0] as the input to _rescaling, so your probe is actually probe[0]. I'm trying to keep my parallelization consistent with the algorithm of original codes.

Reply:
That's a good goal, but in this case, I think my original implementation was incorrect for multi-gpu. If we don't rescale `probe[0:ngpu]`, then we will have the problem you just fixed (the probes not being the same for all nodes/devices). Do you agree?

Reply:
I see that the broadcasting was happening in a place I didn't expect, so I moved it.